### ISSUE TITLE
Wrong NULL behaviour of BitAndAgg, BitOrAgg, and related aggregate function emulations

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#14321]
The current emulation of `bitAndAgg()` and related bitwise aggregate functions is wrong when it comes to producing `NULL` values for empty aggregate sets. E.g. this test should pass:

```java
Record2<Integer, Integer> r2 =
create().select(
            bitAndAgg(TBook_ID().plus(8)),
            bitOrAgg(TBook_ID().plus(8)))
        .from(TBook())
        .where(falseCondition())
        .fetchSingle();

assertNull(r2.value1());
assertNull(r2.value2());
```

But fails for emulations, where the bitwise aggregation looks something like:

```sql
select
  (case
    when (min(case
      when bitand(
        ("informix"."t_book"."id" + 8),
        1
      ) <> 0 then 1
      else 0
    end) = 1) then 1
    else 0
  end + case
    when (min(case
      when bitand(
        ("informix"."t_book"."id" + 8),
        2
      ) <> 0 then 1
      else 0
    end) = 1) then 2
    else 0
  end + ...
```

The current result is:

```
+-----------+----------+
|bit_and_agg|bit_or_agg|
+-----------+----------+
|          0|         0|
+-----------+----------+
```

**Comment by lukaseder** [comment]
This would work better:

```sql
select
  case min(case when bitand(id, 1) <> 0 then 1 else 0 end) 
    when 0 then 0 when 1 then 1 
  end + 
  case min(case when bitand(id, 2) <> 0 then 1 else 0 end)
    when 0 then 0 when 1 then 1
  end + ...
from t_book
where 1 = 0;
```

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0, 3.17.6 (https://github.com/jOOQ/jOOQ/issues/14322), and 3.16.12 (https://github.com/jOOQ/jOOQ/issues/14323)
